static/js/components/dialogs/CollectionFormDialog.js

```diff
@@ -109,25 +76,30 @@ class CollectionFormDialog extends React.Component {
     dispatch(uiActions.setAdminLists(event.target.value));
   };
 
-  submitForm = () => {
+  submitForm = async () => {
     const {
       dispatch,
       hideDialog,
-      collectionUi: { collectionForm }
+      collectionUi,
     } = this.props;
 
-    let patchData = {
+    const collectionForm = getCollectionForm(collectionUi);
+    const payload = {
       title: collectionForm.title,
       description: collectionForm.description,
       view_lists: this.calculateListPermissionValue(collectionForm.viewChoice, collectionForm.viewLists),
       admin_lists: this.calculateListPermissionValue(collectionForm.adminChoice, collectionForm.adminLists)
     };
-    dispatch(actions.collections.patch(collectionForm.key, patchData)).then(
-      (collection) => {
-        hideDialog();
-        this.initializeFormWithCollection(collection);
-      }
-    );
+
+    if (collectionUi.isNew) {
+      await dispatch(actions.collectionsList.post(payload));
+      hideDialog();
+      dispatch(uiActions.initCollectionForm(makeInitializedForm()));
```

Comment:
I think it's better to just create a new action `CLEAR_COLLECTION_FORM` which would set the form to the initial state.